watchtower-plugin/src/retrier.rs

```diff
@@ -59,9 +104,20 @@ impl RetryManager {
                         .contains_key(&tower_id)
                     {
                         log::info!("Skipping retrying abandoned tower {}", tower_id);
-                        continue;
+                    } else if let Some(retrier) = self.retriers.get(&tower_id) {
+                        if retrier.is_iddle() && locators.len() > 1 {
+                            // This can only happens if the tower was manually retried while ideling, given otherwise
+                            // the locators length will be one, or the retrier won't be found in retriers map.
+                            // In that case, the data is already present in the [Retrier], so we just to change the state
+                            log::info!(
+                                "Manually finished idling. Flagging {} for retry",
+                                retrier.tower_id
+                            );
+                            retrier.set_status(RetrierStatus::Stopped)
+                        }
```

Comment:
This is tricky. I may need to explain the logic a bit better.

Appointments used to be passed to the retrier one by one, which was optimal for data that is missed on creation (`on_commitment_revocation`) but suboptimal for data that came either from a bootstrap or a manual retry (in which case we needed to send many items to the retrier one by one using the same `tower_id`).

This has been updated in this PR. Now we send data always in a `(tower_id, HashSet<locator>)` pair, meaning that, if data comes from `on_commitment_revocation` the set will have a single entry, while if this comes from either a bootstrap or a manual retry it will have, potentially, several entries.

Then we build the logic around wether the retrier is `Idle` and what is the length of the received set:

- If data comes from a bootstrap, the retrier cannot exist, therefore we call ` self.add_pending_appointments(tower_id, locators)` which, in turn, will add a new retrier with all the locators.
- If data comes from a manual retry there are two cases:
    - The retrier is `Idle`, in which case the data is already in memory, we just need to update the state to force a retry
    - The retrier is not `Stopped`, in which case is not part of retriers so we default to the same case as for the bootstrap
-  If data comes from a channel update (`on_commitment_revocation`), then the set's length is one, so we **DO** need to add this to pending.

However, after explaining all this I just realized that there is an edge case that is not covered. If there is a manual retry with a single pending appointment (which is rare but I guess it could happen), this will be seen as data coming from a channel update, and the retry won't be forced. Data won't be replicated either, given the element will already be in the `pending_appointments` set for the corresponding retrier, but the plugin won't react as expected.

I guess I'll need to add a custom type based on an enum that helps distinguish where this comes from, which is exactly what I was trying to avoid being clever with the retrier state and the length of the set 😅

Reply:
Fixed this in [3750d24](https://github.com/talaia-labs/rust-teos/pull/168/commits/3750d249ff04be293230426566fd775dd415a833). Data now carries around whether it is fresh or stale, which help with the decision making of the aforementioned cases without having to be clever about the size of the received data

Reply:
@mariocynicys I was rechecking this while adding some tests and `self.add_pending_appointments` is indeed needed on an else clause for `if retrier.is_idle() && data.is_stale()` to cover fresh data. Is that what you referred to?

Reply:
> Is that what you referred to?

Yup that's it.
But you didn't add it yet right? cuz i don't see the other branch in the diff.
And i think it should work if we just do `self.add_pending_appointment` in any case (other than for an abandoned tower).

Reply:
I haven't pushed these changes yet, working on adding the ones suggested in discord and updating the tests.

Regarding this specifically, I think we should add them in all cases except if the tower is abandoned and when the data is stale and the retrier is idle.

Reply:
> I haven't pushed these changes yet

Cool then.

> except if ... and when the ***data is stale***

Errmm, not really sure why but I think I will get this with another round of review.

Reply:
It should be there now